sdk/ai/ai-projects/src/agents/assistantsTrace.ts

```diff
@@ -0,0 +1,44 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+import { TracingSpan } from "@azure/core-tracing";
+import { AgentOutput } from "../generated/src/outputModels.js";
+import { TracingAttributeOptions, TracingAttributes, TracingUtility, TrackingOperationName } from "../tracing.js";
+import { CreateAgentParameters } from "../generated/src/parameters.js";
+import { addInstructionsEvent, formatAgentApiResponse } from "./traceUtility.js";
+
+/**
+ * Traces the start of creating an agent.
+ * @param span - The span to trace.
+ * @param options - The options for creating an agent.
+ */
+export function traceStartCreateAgent(span: Omit<TracingSpan, "end">, options: CreateAgentParameters): void {
```

Comment:
Is this meant to be reusable and / or something for customers? If not, why don't we add these to the span directly in the createAgent call? Something similar to:

 ```ts
tracingClient.withSpan("name", async (updatedOptions, span) => {
   span.setAttribute(...) // access to the span here
   const result = await context.path("/assistants").post(updatedOptions);
});
```

etc. Would that not work? TracingClient#withSpan provides the callback with a second argument which is the span itself

Reply:
Want to keep the code clean and provide wrapping functionality that sets attributes before and after the call.